csclient/params/params.go

```diff
@@ -258,6 +262,30 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource holds the metadata about a Resource defined for a charm.
+type Resource struct {
+	Name        string
+	Type        ResourceType
```

Comment:
That seems like an argument for a helper then (e.g. the one we already have). :)